shuffle-plugin/pom.xml

```diff
@@ -124,6 +124,12 @@
                     <version>${spark.version}</version>
                     <scope>provided</scope>
                 </dependency>
+	        <dependency>
+                    <groupId>com.google.guava</groupId>
```

Comment:
Why are we adding a guava dependency only in the databricks profile?  Seems like we're missing a general guava dependency instead.  Applies to the sql-plugin guava dependency as well.  It looks like we're only using guava for `ThreadFactoryBuilder` which we may want to reconsider, as guava is a notoriously messy dependency in practice.